lib/TippyTopProvider.js

```diff
@@ -0,0 +1,67 @@
+/* globals require, JSON */
+"use strict";
+
+const data = require("sdk/self").data;
+const URL = require("sdk/url").URL;
+
+const DEFAULT_OPTIONS = {
+  sites: null
+};
+
+function TippyTopProvider(options = {}) {
+  this.options = Object.assign({}, DEFAULT_OPTIONS, options);
+  this.init();
+}
+
+TippyTopProvider.prototype = {
+  /**
+   * Initialize the sites object
+   */
+  init() {
+    // If the sites were passed into the constructor (for testability), use those.
+    // Else, read them from top_sites.json.
+    let sites;
+    if (this.options.sites) {
+      sites = this.options.sites;
+    } else {
+      sites = JSON.parse(data.load("content/favicons/top_sites.json"));
```

Comment:
How do I do that? Do I need to copy over that index.js into the addon data directory? Can I use it from there with require? Since it's in node_modules, wasn't sure how.